### ISSUE TITLE
Add bulk update condition from See All Collection Objects in This Container

### ISSUE THREAD

**Issue by campmlc** [ArctosDB/arctos#4987]
In Object Tracking, we currently have a super useful tool to be able to:
1) Go to Manage Data > Object Tracking > Find Container
2) Click on box to pull up Container Details
3) Click on See All Collection Objects in this Container at right
4) Scroll all the way down to end of display, and Add All Items to Loan and Update Part Disposition to On Loan.

This is extremely helpful when we have all loan items scanned into a container- everything can be added a loan with one click.
I would like to request to add to this form the ability to Update Part Condition.
So in addition to "Update Part Disposition", we would have a second field that would allow us to Update Part Condition for all parts in this container.
Reason: This would allow bulk update of part condition for all tissue samples in a dewar that went dry or a freezer that went down, regardless of part name etc.
Alternatives: 
1) Use Manage > Modify Parts tool in Catalog Record Search results. But this works at the level of the catalog record, not at the level of the container, and it must be done separately for every part name, and it is not possible to verify part location from the form, to confirm all are in the affected container.
2) Some other bulkloader that would allow for updating part condition from a barcode csv list. This may also be useful regardless for large csv files, such as a downed freezer. Ideally, we would have both tools?

Priority: high - needing to get a grip on the ongoing problem of managing and tracking changes to tissue quality.

**Comment by dustymc** [comment]
"managing and tracking changes to tissue quality" was addressed in https://github.com/ArctosDB/arctos/issues/4535

**Comment by campmlc** [comment]
Would it be that difficult to enable this regardless? This is a simple and logical place to go to make updates for the same reason it is useful to add objects to loan. No need to use a loader. Just find the container and update the child barcodes. 
FWIW I have no documentation on #4535.

**Comment by campmlc** [comment]
So the part attribute solution does not work in this case because I need to update condition. Right now we have two different ways of saying "condition", which needs to be addressed. But as a start, I would like to re-iterate my request to have the tool above allow for bulk updates of part condition by container.
The example I am currently working with is a LN2 dewar of tissues that went dry for several weeks. The tissues were removed and put into freezer boxes and then brought to the museum, where they were scanned. The tissue parts, however, were cataloged with the rest of the accession and given "condition = 5" by the division, presumably because tissues were field collected directly into nitrogen. However, these particular tissues are not "5" - they should be recorded as 1=poor. 
If I add a part attribute for tissue quality of poor, that will directly contradict the condition of "5", and these two types of data are located in different places and in different contexts in the UI. I need the ability to update condition to reflect the current condition. Unless we can set condition to update automatically when a condition related part attribute is uploaded, I request a tool that will allow this, as above.

**Comment by campmlc** [comment]
@dustymc

**Comment by campmlc** [comment]
Just a further note: we have a variety of tools that allow for updating of part disposition, the above being just one example. But to my knowledge there is no tool that allows bulk updating of condition. If Find Container tool above is not acceptable, then we should have a bulkloader that will allow updated condition to a csv list of barcodes. This latter will be more generally useful because they objects do not all have to occupy the same container. However, I still maintain that adding this tool along with Update Part Disposition in the above tool makes sense from a usability and UI perspective, and given that the condition of many parts is dependent on their container environments.

**Comment by Jegelewicz** [comment]
@campmlc I believe you can change the condition for parts within a collection using the Modify Parts tool from search results.

<img width="960" alt="Screenshot 2022-08-30 08 14 51" src="https://user-images.githubusercontent.com/5725767/187460681-2fbd1159-ceac-405b-ad01-8c202da15047.png">

![image](https://user-images.githubusercontent.com/5725767/187460455-8f92effb-643e-4e91-a41e-7bab81fb6b52.png)

The barrier I see is when you have two identical parts in a record. So I do see the need for selecting parts either by container or part identifier for this kind of update. But, for cases where this isn't a problem, this could save you some time?

**Comment by Jegelewicz** [comment]
OK so now I see that you cannot modify condition - why?

**Comment by campmlc** [comment]
Yes, but you can only modify one part type at a time. So you can only do
all livers, Then all hearts, Then all lungs, Then all spleens. Etc. And you
can't see or confirm that the parts you are modifying are actually in the
affected containers from that tool.
So that does not solve the problem.
The solution I am proposing does, and is already in place for exactly the
same purpose for changing disposition.

**Comment by dustymc** [comment]
> Right now we have two different ways of saying "condition",

Nope.

1. [Condition](https://handbook.arctosdb.org/documentation/parts.html#condition) is the attached "fact." (Not entirely - there's a little metadata, but it's neither controllable nor very exposed).
2. [condition report](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecpart_attribute_type#condition_report) is basically condition, but without the structural limitations (eg you can fully expose how things change over time) and with full metadata
3. [preservation](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctspecpart_attribute_type#preservation)
4. preservation need
5. remaining volume
6. storage temperature
7. tissue quality

and probably more. Plain ol' condition can't quite be _wrong_ here, but it's a very general-purpose concept - a fact acknowledged by the existence of the sea of better-for-situation alternatives. Two of those alternatives can be combined to perfectly describe the situation: storage temperature changed causing tissue quality to change.

>different places and in different contexts in the UI.

That I'll buy. File an Issue (I think a first step is as simple as adding condition to the 'part' search - maybe its already there - but I'd got for something radical in https://github.com/ArctosDB/arctos/issues/4707 or getting rid of it altogether or displaying it with attributes or....)

>If I add a part attribute for tissue quality of poor, that will directly contradict the condition of "5",

Yes, that is fully expected and supported, and I've been pointing it out (mostly in the context of things like summaries) for some time. I'm still not convinced that we have a lot of users who aren't capable of understanding that things change over time, and so I'm not at all convinced that these data conflict in any way.

If you absolutely must use what still looks to me like the least appropriate not-wrong approach, there's a tool in the usual place, and the part attribute tool could be modified to write to it.

<img width="523" alt="Screen Shot 2022-08-30 at 8 35 25 AM" src="https://user-images.githubusercontent.com/5720791/187480189-996a25fe-bfdd-480e-9714-e86f1c43ec43.png">

**Comment by campmlc** [comment]
Ok, so it sounds like it is impossible to accommodate my request and make an easily usable tool which enables what I need with one click after reviewing a container's contents (and I don't understand the reluctance to add this, since one of the really neat things about Arctos is that there are these cool shortcuts that allow you to, for example, add all parts from a container to a loan. I don't see the difference here. But I can see that perhaps that creates maintenance problems down the line.)
In that case, in order to use the component bulkloader to do the same thing, I need to request that the Parts >Bulk Edit tool have the field "barcode" added to the list of field options, so I don't have to do multiple time consuming steps to convert from known barcodes to unknown part IDs and risk data screw ups in the process. Pretty please?

**Comment by dustymc** [comment]
>impossible to accommodate 

I don't know how you got that from what I said.

>Parts >Bulk Edit tool have the field "barcode" added 

There is clearly some fundamental misunderstanding here. Barcode is not an unambiguous part handle, and that tool cannot function without.

**Comment by Jegelewicz** [comment]
K - removed Nanograms per milliliter. A concentration measurement used when working with DNA. <a href="https://github.com/ArctosDB/arctos/issues/3624#issuecomment-1212128226" class="external">GitHub Issue</a> from the table - I think a ratio table would be good and we could move percent there? - I'll start a new issue.

**Comment by campmlc** [comment]
Yes, I think that #4707 may be a good solution. One related question is - what field are we publishing to GGBN for their resource for tissue quality or condition? Whatever we do here, we should make sure we update what we publish to them.

**Comment by dustymc** [comment]
> what field are we publishing to GGBN

We're not at the moment, https://github.com/ArctosDB/arctos/issues/3699

**Comment by campmlc** [comment]
Not good.

**Comment by campmlc** [comment]
In order to keep track of all this without dropping the ball, here are the container contents downloads of the two boxes of tissues that came from the dewar that went dry. Note that I only have container IDs and barcodes for the containers of these parts, not part IDs - so that's the reason I can't use part bulk edit to change condition. This is the reason for my original request starting this thread. 
In the catalog record, all these parts are recorded as Condition = 5, excellent, but these tissues were all in a dewar which defrosted for several weeks before reaching the museum. I assume that when Condition is changed to a part attribute, Condition =5 will get converted to part attribute ="condition report" = "5", and it will get some arbitrary date on the date of conversion? So do I need to wait until that is done before uploading a updated condition, so that it reflects the current condition? I don't want the legacy conversion to post-date the "current" status? Advice? @dustymc 

[container_contents_download_DGR11455.zip](https://github.com/ArctosDB/arctos/files/9483527/container_contents_download_DGR11455.zip)


[container_contents_download_DGR22921.zip](https://github.com/ArctosDB/arctos/files/9483528/container_contents_download_DGR22921.zip)

**Comment by dustymc** [comment]
> not part IDs 

<img width="168" alt="Screen Shot 2022-09-06 at 1 25 27 PM" src="https://user-images.githubusercontent.com/5720791/188731610-406921be-affc-482c-bfea-7810b088458e.png">


<img width="362" alt="Screen Shot 2022-09-06 at 1 25 43 PM" src="https://user-images.githubusercontent.com/5720791/188731636-1c937100-53f2-4e90-be41-59ca226b724a.png">

<img width="530" alt="Screen Shot 2022-09-06 at 1 26 08 PM" src="https://user-images.githubusercontent.com/5720791/188731648-dc51da10-ec9a-47ab-bedf-3507d392f90a.png">

>In the catalog record, all these parts are recorded as Condition = 5

Which means nothing to anyone but whoever created that, at best. (Everyone else will assume it's a misplaced count or a cat was wandering around on your keyboard or something).

> that when Condition is changed to a part attribute, Condition =5 will get converted to part attribute ="condition report" = "5", 

Two things:

1. There is an [existing non-cryptic way to provide that information ](https://arctos.database.museum/info/ctDocumentation.cfm?table=cttissue_quality). Condition report is free-text and you can stuff whatever you want in there, but there is extremely limited utility in doing so.
2. The better options all exist now, you can use them now, there's absolutely no reason to wait on anything before using them. MAYBE condition='see part attributes' or something would be a sort of best practice, but nobody's going to get confused by better data being available. 

>will get some arbitrary date 

If you use the tools, it'll get whatever you specify.

When/if we eventually get around to merging condition into the attribute, it'll get whatever you tell me. (Which won't matter in the slightest if you start providing better data now.)

>uploading a updated condition

Again, I am in no way recommending this, I don't think it can be see as anything other than tossing out information, but that part attribute CSV-maker above contains part IDs and you can use them to bulk-edit parts.
